### PR TITLE
107/update interface

### PR DESCRIPTION
**Pull request by xiaotiansu**

### CODE REVIEW

File: src/components/space/charts-layout/ChartsHeader.js [hunk 1]
```diff
@@ -36,21 +39,23 @@ const useStyles = makeStyles((theme) => ({
...
     paddingBottom: theme.spacing(2),
   },
   alert: {
     display: 'flex',
     alignItems: 'center',
     marginTop: theme.spacing(1),
   },
+  tmp: {
+    paddingTop: theme.spacing(10),
+  },
```
Comment (pyphilia) [inline src/components/space/charts-layout/ChartsHeader.js:52]: Remove?

File: src/components/space/charts/ActionsByDayChart.js [hunk 2]
```diff
@@ -27,26 +28,48 @@ const useStyles = makeStyles(() => ({
 
 const ActionsByDayChart = () => {
   const { t } = useTranslation();
-  const theme = useTheme();
+  // const theme = useTheme();
```
Comment (pyphilia) [inline src/components/space/charts/ActionsByDayChart.js:31]: Remove?

File: src/components/space/charts/ActionsByDayChart.js [hunk 3]
```diff
@@ -27,26 +28,48 @@ const useStyles = makeStyles(() => ({
...
+  } else if (!noUsers && noActions) {
     actionsByDay = getActionsByDay(filterActionsByUser(actions, selectedUsers));
+  } else if (noUsers && !noActions) {
+    actionsByDay = getActionsByDay(
+      filterActionsByAction(actions, selectedActions),
+    );
+  } else {
+    const filteredByUser = filterActionsByUser(actions, selectedUsers);
+    actionsByDay = getActionsByDay(
+      filterActionsByAction(filteredByUser, selectedActions),
```
Comment (pyphilia) [inline src/components/space/charts/ActionsByDayChart.js:71]:
```suggestion
filterActionsByActionType(filteredByUser, selectedActions),
```

File: src/components/space/charts/ActionsByTimeOfDayChart.js [hunk 4]
```diff
@@ -38,17 +40,39 @@ const ActionsByTimeOfDayChart = () => {
...
-  let actionsByTimeOfDay;
-  if (
+  const groupBy = (key, arr) =>
+    arr.reduce(
+      (acc, cur) => ({
+        ...acc,
+        [cur[key]]: cur[key] in acc ? acc[cur[key]].concat(cur) : [cur],
+      }),
+      {},
+    );
```
Comment (pyphilia) [inline src/components/space/charts/ActionsByTimeOfDayChart.js:50]: I think you can create a new file in `utils`, like `array.js` where you add this function.

File: src/components/space/charts/ActionsByTimeOfDayChart.js [hunk 5]
```diff
@@ -38,17 +40,39 @@ const ActionsByTimeOfDayChart = () => {
...
     );
+  } else if (noUsers && !noActions) {
+    actionsByTimeOfDay = getActionsByTimeOfDay(
+      filterActionsByAction(actions, selectedActions),
+    );
+  } else {
+    const filteredByUser = filterActionsByUser(actions, selectedUsers);
+    actionsByTimeOfDay = getActionsByTimeOfDay(
+      filterActionsByAction(filteredByUser, selectedActions),
+    );
```
Comment (pyphilia) [inline src/components/space/charts/ActionsByTimeOfDayChart.js:75]: Looks like you do that often, probably you can create a util function as well `fitterActions = ({users, types}) => {...}`

File: src/utils/api.js [hunk 6]
```diff
@@ -173,6 +173,13 @@ export const filterActionsByUser = (actions, usersArray) => {
   );
 };
 
+export const filterActionsByAction = (actions, actionsArray) => {
```
Comment (pyphilia) [inline src/utils/api.js:176]: Rename with `filterActionsByActionTypes`

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice PR! I've left some comments, let me know if it's not clear! 🦩

File: src/utils/api.js [hunk 7]
```diff
@@ -173,6 +173,13 @@ export const filterActionsByUser = (actions, usersArray) => {
   );
 };
 
+export const filterActionsByActionTypes = (actions, actionsArray) => {
+  const actionKey = 'actionType';
+  return actions.filter((action) =>
+    actionsArray.some((act) => act.value === action[actionKey]),
```
Comment (pyphilia) [inline src/utils/api.js:179]:
```suggestion
actionsArray.some((act) => act.value === action.actionType)
```

**Review by pyphilia — APPROVED** [review]
Really good, I've added a minor suggestion, but after  that you can merge!

File: src/utils/api.js (hunk 7 above)
Comment (xiaotiansu) [reply src/utils/api.js:179]: I followed the convention of the previous function, then should I also change the previous one?
image

**Pull request closed without merging**